api/v1/region_types.go

```diff
@@ -15,10 +16,15 @@ const (
 type QdrantCloudRegionSpec struct {
 	// Id specifies the unique identifier of the region
 	Id string `json:"id,omitempty"`
+	// Components specifies the list of components to be installed in the region
+	// +optional
+	Components []corev1.TypedObjectReference `json:"components,omitempty"`
```

Comment:
I would suggest that we craft our own type instead of using upstream `TypedObjectReference` - it is going to save us headache if something changes upstream and with that our API is going to change.

Reply:
Updated.